ext/openssl/ossl_x509store.c

```diff
@@ -188,6 +188,7 @@ ossl_x509store_set_vfy_cb(VALUE self, VALUE cb)
 
     GetX509Store(self, store);
     X509_STORE_set_ex_data(store, store_ex_verify_cb_idx, (void *)cb);
+    RB_OBJ_WRITTEN(self, Qundef, cb);
```

Comment:
I am comparing the `ossl_x509store_set_vfy_cb` from `ossl_x509stctx_verify`.

In the `ossl_x509store_set_vfy_cb`, the `RB_OBJ_WRITTEN` is called before the `rb_iv_set`. In the `ossl_x509stctx_verify`, the `RB_OBJ_WRITTEN` after the `rb_iv_set`. Is it okay?

Reply:
Yes, `rb_iv_set` doesn't matter here, because when you create a reference using instance variables like this, the write barrier is automatically triggered.

The reason we have to call `RB_OBJ_WRITTEN` here, is because we gave a `VALUE` reference to `X509_STORE_set_ex_data` and it's holding onto it, so we need to tell the GC about that.

Reply:
All right!